### ISSUE TITLE
Trying To Play An Animation With A Function Call Track Corrupts Project

### ISSUE THREAD

**Issue by Kalcen** [godotengine/godot#88654]
### Tested versions

v4.2.1.stable.official [b09f793f5]

### System information

Windows 10

### Issue description

This project no longer opens without instantly crashing. The only way to fix it that I know of is by deleting the `node_3d.tscn` file. I was trying to test a separate issue where function calls weren't being called by animation tracks sometimes, but this happened instead.

### Steps to reproduce

I caused this to happen by taking these steps:
Create a scene with:
-  3D Camera
- Directional Light
- Mesh Instance 3D (displaying a cube)
- Animation Player
Create a simple new animation in the animation player that moves the cube up and down. Add another track that calls a method named `test()` from a script attached to the `MeshInstance3D` node. All the test function should do is print something to the output console. Try to hit play on the animation in the editor, this should crash the project. If you save the scene before doing this, the project will no longer be able to open unless the scene file is deleted.

### Minimal reproduction project (MRP)

[AnimationFunctionCallIssue.zip](https://github.com/godotengine/godot/files/14370487/AnimationFunctionCallIssue.zip)

**Comment by Mrinal96** [comment]
Hey! 
So, I imported your code into Godot and was able to view the cube in the app. But when I click on the play button, it loads and immediately crashes (same issue as yours) . I tried implementing the test function like you instructed by attaching the script to MeshInstance3D but unfortunately, as soon as I click on the play button, it crashes , which goes to show that we might have to delete the node_3d.tscn file. May I know what are you using this .tscn file for?
My MacBook is 16 GB RAM with 2.6 Ghz Intel i7 processor, I dont think its a memory issue .But we can address this further once you let me know why you are using this .tscn file.
<img width="1083" alt="Screenshot 2024-02-22 at 6 29 23 PM" src="https://github.com/godotengine/godot/assets/31739872/7ab8a236-776f-45f0-bd36-655d3c18d147">
